pymeasure/display/console.py

```diff
@@ -0,0 +1,267 @@
+        super().__init__(**kwargs)
+        self.procedure_class = procedure_class
+
+    def setup_parser(self, inputs):
+        """ Setup command line arguments parsing from parameters information """
+
+        self.procedure = self.procedure_class()
+        parameter_objects = self.procedure.parameter_objects()
+        for name in inputs:
+            kwargs={}
+            parameter = parameter_objects[name]
+            default, help_fields,_type = parameter.cli_args
+            kwargs['help'] = self._cli_help_fields(parameter.name, parameter, help_fields)
+            kwargs['default'] = default
+            if _type != None:
+                kwargs['type'] = _type
+            self.add_argument("--"+name, **kwargs)
+
+        special_options = copy.deepcopy(self.special_options)
+        for option, kwargs in special_options.items():
+            help_fields = [('units are', 'units')] + kwargs['help_fields']
+            desc = kwargs['desc']
+            kwargs['help'] = self._cli_help_fields(desc, kwargs, help_fields)
+            del kwargs['help_fields']
+            del kwargs['desc']
+            self.add_argument("--" + option, **kwargs)
+
+    def _cli_help_fields(self, name, inst, help_fields):
+        def hasattr_dict(inst, key):
+            return key in inst
+        def getattr_dict(inst, key):
+            return inst[key]
+
+        if isinstance(inst, dict):
+            hasattribute = hasattr_dict
+            getattribute = getattr_dict
+        else:
+            hasattribute = hasattr
+            getattribute = getattr
+
+        message = name
+        for field in help_fields:
+            if isinstance(field, str):
+                field = ["{} is".format(field), field]
+
+            if hasattribute(inst, field[1]) and getattribute(inst, field[1]) != None:
+                prefix = field[0]
+                value = getattribute(inst,field[1])
+                message += ", {} {}".format(prefix, value)
+
+        message = message.replace("%", "%%")
+        return message
+
+class ManagedConsole(QtCore.QCoreApplication):
+    """
+    Base class for console experiment management .
+
+    Parameters for :code:`__init__` constructor.
+
+    :param procedure_class: procedure class describing the experiment (see :class:`~pymeasure.experiment.procedure.Procedure`)
+    :param inputs: list of :class:`~pymeasure.experiment.parameters.Parameter` instance variable names, which the display will generate graphical fields for
+    :param log_channel: :code:`logging.Logger` instance to use for logging output
+    :param log_level: logging level
+    :param sequence_file: simple text file to quickly load a pre-defined sequence with the :code:`Load sequence` button
+    :param directory_input: specify, if present, where the experiment's result will be saved.
+    """
+    def __init__(self,
+                 args,
+                 procedure_class,
+                 inputs=(),
```

Comment:
@CasperSchippers, I originally assumed that `input` parameter was to select parameters that user can change, but now I realize you may be right. This may be used to avoid to limit space in the UI, so I would vote to remove the parameter.